okhttp-tests/src/test/java/com/squareup/okhttp/ConnectionPoolTest.java

```diff
@@ -187,16 +179,46 @@ private void awaitGarbageCollection() throws InterruptedException {
     System.runFinalization();
   }
 
-  private RealConnection newConnection(ConnectionPool pool, Route route, long idleAtNanos) {
-    RealConnection connection = new RealConnection(route);
-    connection.idleAtNanos = idleAtNanos;
-    connection.socket = new Socket();
+  private RealConnection newIdleConnection(ConnectionPool pool, Route route, long idleAtNanos) {
+    RealConnection connection = createStubbedConnection(route);
     synchronized (pool) {
+      connection.idleAtNanos = idleAtNanos;
       pool.put(connection);
     }
     return connection;
   }
 
+  private RealConnection newActiveConnection(ConnectionPool pool, Route route,
+      StreamAllocation streamAllocation) {
+    RealConnection connection = createStubbedConnection(route);
+    synchronized (pool) {
+      streamAllocation.acquire(connection);
+      pool.put(connection);
+    }
+    return connection;
+  }
+
+  private RealConnection newLeakedConnection(ConnectionPool pool, Route route) {
+    RealConnection connection = createStubbedConnection(route);
+    synchronized (pool) {
+      allocateAndLeakAllocation(pool, connection);
+      pool.put(connection);
+    }
+    return connection;
+  }
+
+  /** Use a helper method so there's no hidden reference remaining on the stack. */
+  private void allocateAndLeakAllocation(ConnectionPool pool, RealConnection connection) {
```

Comment:
Inline this? The comment about the stack will apply to `newLeakedConnection()` also